diff --git a/src/hsm/PSFCorr.cpp b/src/hsm/PSFCorr.cpp
index a9b3260..21b92b1 100644
--- a/src/hsm/PSFCorr.cpp
+++ b/src/hsm/PSFCorr.cpp
@@ -728,7 +728,9 @@ namespace hsm {
         if (iy2 > ymax) iy2 = ymax;
         dbg<<"y1,y2 = "<<y1<<','<<y2<<std::endl;
         dbg<<"iy1,iy2 = "<<iy1<<','<<iy2<<std::endl;
-        assert(iy1 <= iy2);
+        if (iy1 > iy2) {
+             throw HSMError("Vertical data bounds (iy1,iy2) don't make sense.");
+        }
 
         //
         /* Use these pointers to speed up referencing arrays */
