diff --git a/ups/galsim.table b/ups/galsim.table
index 974ac48..861668f 100644
--- a/ups/galsim.table
+++ b/ups/galsim.table
@@ -7,5 +7,6 @@ setupRequired(tmv)
 envPrepend(PYTHONPATH, ${PRODUCT_DIR})
 envPrepend(LD_LIBRARY_PATH, ${PRODUCT_DIR}/lib)
 envPrepend(DYLD_LIBRARY_PATH, ${PRODUCT_DIR}/lib)
+envPrepend(LSST_LIBRARY_PATH, ${PRODUCT_DIR}/lib)
 envPrepend(PYTHONPATH, ${PRODUCT_DIR}/lib/python)
 envPrepend(PATH, ${PRODUCT_DIR}/bin)
